makefiles/vars.inc.mk

```diff
@@ -5,6 +5,8 @@ export Q                     # Used in front of Makefile lines to suppress the p
 export QQ                    # as Q, but be more quiet
 export QUIET                 # The parameter to use whether to show verbose makefile commands or not.
 
+export OS                    # The operating system of the build host
```

Comment:
Can we get away with a conditional export? So its not exported for all targets but only when using `jerryscript`.

Reply:
IMHO the `OS` variable might be valuable in general during build process, at least if we plan to support more OSes properly - other than just Linux. Also I wouldn't introduce conditionals in this file and moving this export somewhere else breaks consistency and makes it harder to find where variables are exported. I think its good to aim for having one place (i.e. file) to make explicit exports.

Reply:
I wasn't talking about exporting it here, I would have exported it only in the `jerryscript` Makefile and only for the required target. 

So far `jerryscript` is the first use of `OS` in a `subshell` and the variable is evaluated at the start of `Makefile.include` so present almost everywhere except in `subshells`.  If you think it will be thoughtfully needed to extended OS support then lets export it, but if possible change the commit message so it reflects the reason why it is being exported.

I would still prefer having:

```
all: export OS=$(OS)
	"$(MAKE)" -C $(PKG_BUILDDIR) -f $(CURDIR)/Makefile.jerryscript all
```

or

```
$(call target-export-variables,all,OS)
all:
	"$(MAKE)" -C $(PKG_BUILDDIR) -f $(CURDIR)/Makefile.jerryscript all
```

as long as the need for the export is not widespread.

But I won't insist on this, I trust your judgement on the need for the variable to support other `OS`, so if you add a more descriptive comment to the commit I'm happy whether you keep the export or not (something like `Export the OS variable since it should be needed for proper multi-OS support. Currently only needed by 'pkg/jerryscript'`).

Reply:
> But I won't insist on this,

Plus `OS = $(word 1, $(UNAME))` wouldn't cause much overhead if exported, so I think I'm just being overzealous (the commit message addition would still be appreciated).

Other than that changes are good and make sense, so feel free to squash.